database/migrations/000073_sdlc_entities.down.sql

```diff
@@ -0,0 +1,13 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
```

Comment:
Should we have some sort of down migration where we remove the new values?

Reply:
Any idea what it would look like? I think you cannot drop values from the enums

Reply:
I think we should add a down migration anyway, it would look like this
```sql
BEGIN;

CREATE TYPE entities_revert AS ENUM ('repository', 'build_environment', 'artifact', 'pull_request');
DELETE table1 WHERE column1 IN ('release', 'pipeline_run', 'task_run', 'build');
ALTER TABLE table1 ALTER COLUMN column1 TYPE entities_revert USING column1::text::entities_revert;
DROP TYPE entities;
ALTER TYPE entities_revert RENAME TO entities;

COMMIT;
```
The `ALTER TABLE` statement would fail in case data already exists with the new enum values that are not be supported by the old version, preventing the system from reverting the change. The `DELETE` can be omitted if one accepts to try a revert, have it fail, and thus fix the problem by introducing a followup change.

I understand and sort of agree that it's not worth it given we're not currently using it, but then it might be better to move migrations to a follow up PR along with changes involving them unless it is strictly necessary to have them available.